codegen/gateway.go

```diff
@@ -650,3 +749,16 @@ func (gateway *GatewaySpec) GenerateMain() error {
 	)
 	return err
 }
+
+// GenerateMiddlewareSchemas will generate the middleware schema files
+// for the gateway
+func (gateway *GatewaySpec) GenerateMiddlewareSchemas() error {
```

Comment:
Refactored to use a schema instead of trying to generate one so we can rely on external packages, as you recommended. PTAL.